mutation_reader.hh

```diff
@@ -35,20 +36,20 @@ class reader_selector {
 protected:
     schema_ptr _s;
 private:
-    dht::ring_position_view _selector_position;
+    optimized_optional<dht::ring_position_view> _selector_position;
 protected:
```

Comment:
I think this is introduces complications. ring_position_view should be able to represent the every bound that partition_range can represent.

One way to do that is to allow ring_position_view::_token to be null. Another way to to set it to the maximum numeric token (we can do that since we removed byte_ordered_partitioner), with _weight = 1. Probably the second way is better in terms of regression risk since it just sets the token to a value.

Reply:
Actually, a partition_range bound is similar to optional<ring_position>, so making ring_position optional is the right thing. It introduces complications but fewer in the long run. So I think this is the correct direction.